analysis_notebooks/Geography Analysis.ipynb

```diff
@@ -306,154 +311,253 @@
+    "        if country in counts.keys():\n",
+    "            counts['United+Kingdom'] += counts[country]\n",
+    "        counts.pop(country, None)\n",
+    "\n",
+    "    for country, count in sorted(counts.items(), key=lambda x: x[1], reverse=True):\n",
+    "        country_count_phrases.append(\"{0}: {1}   {2}%\".format(country, count, round(100.*count/total)))\n",
+    "        iso3 = country_to_iso3(country)\n",
+    "        if iso3 is not None:\n",
+    "            iso3_to_bin[iso3] = np.digitize(count, bins)\n",
+    "            try:\n",
+    "                iso3_to_scaledpop[iso3] = count / CountryInfo(country.replace('+', ' ')).population()\n",
+    "            except KeyError:\n",
+    "                pass\n",
+    "    #             print(\"{} not found in CountryInfo\".format(country))\n",
     "        try:\n",
-    "            iso3_to_scaledpop[iso3] = count / CountryInfo(country.replace('+', ' ')).population()\n",
+    "            subregion = iso3_to_subregion[iso3]\n",
+    "            if subregion in subregion_counts.keys():\n",
+    "                subregion_counts[subregion] += count\n",
+    "            else:\n",
+    "                subregion_counts[subregion] = count\n",
     "        except KeyError:\n",
-    "            pass\n",
-    "#             print(\"{} not found in CountryInfo\".format(country))\n",
-    "    try:\n",
-    "        subregion = iso3_to_subregion[iso3]\n",
-    "        if subregion in subregion_counts.keys():\n",
-    "            subregion_counts[subregion] += count\n",
-    "        else:\n",
-    "            subregion_counts[subregion] = count\n",
-    "    except KeyError:\n",
-    "        print(\"This country's subregion not found: {}\".format(country))\n",
+    "            print(\"This country's subregion not found: {}\".format(country))\n",
+    "    for key in iso3_to_scaledpop.keys():\n",
+    "        iso3_to_scaledpop[key] /= min(iso3_to_scaledpop.values())\n",
     "\n",
-    "for key in iso3_to_scaledpop.keys():\n",
-    "    iso3_to_scaledpop[key] /= min(iso3_to_scaledpop.values())\n",
+    "\n",
+    "else:\n",
+    "    counts_gps = pickle.load(open(\"results/{}/geo_ctr_gps.pkl\".format(folder_name), \"rb\"))\n",
+    "    region_to_id = counts_gps[\"region_to_id\"]\n",
+    "    id_to_gps = counts_gps[\"id_to_gps\"]\n",
+    "    geo_boundaries = dataset.geo_boundaries\n",
+    "    choro_data = pd.read_csv(dataset.choropleth_filepath)\n",
+    "\n",
+    "    counts = {}\n",
+    "    for region in region_to_id:\n",
+    "        counts[region] = len(region_to_id[region])\n",
+    "    total = sum(counts.values())\n",
+    "    country_count_phrases = []\n",
+    "\n",
+    "\n",
+    "    for country, count in sorted(counts.items(), key=lambda x: x[1], reverse=True):\n",
+    "        country_count_phrases.append(\"{0}: {1}   {2}%\".format(country, count, round(100.*count/total)))\n",
+    "\n",
+    "    colors = [\n",
```

Comment:
what happens if there's more things to visualize than colors?

Reply:
have it currently so that it wraps by modding the total number of available colors. like this:`color=colors[count%len(colors)]` 

is there another way that would work better?

Reply:
discussed